src/java/org/apache/cassandra/service/snapshot/TableSnapshot.java

```diff
@@ -138,11 +152,21 @@ public boolean isExpiring()
 
     public long computeSizeOnDiskBytes()
     {
-        return snapshotDirs.stream().mapToLong(FileUtils::folderSize).sum();
+        long sum = sizeOnDisk;
+        if (sum == 0)
+        {
+            for (File snapshotDir : snapshotDirs)
+                sizeOnDisk = sum += FileUtils.folderSize(snapshotDir);
+        }
```

Comment:
How can the race happen querying the vtable when the version of `getSnapshots` which is eventually called in SnapshotManager is `synchronized`?

Reply:
And yes, we can make it `volatile`.

Reply:
It think it would be the best if we just made that method `synchronized`. This method is supposed to be called just once anyway because it is not counting it all over again, just once, when sum is 0.